server/blob/blob-cassandra/src/main/java/org/apache/james/blob/cassandra/cache/CachedBlobStore.java

```diff
@@ -68,8 +68,11 @@ public InputStream read(BucketName bucketName, BlobId blobId) throws ObjectStore
             .filter(defaultBucket::equals)
             .flatMap(ignored ->
                 Mono.from(cache.read(blobId))
-                    .<InputStream>flatMap(bytes -> Mono.fromCallable(() -> new ByteArrayInputStream(bytes))))
-            .switchIfEmpty(Mono.fromCallable(() -> backend.read(bucketName, blobId)))
+                    .flatMap(this::toInputStream)
+                    .switchIfEmpty(readFromBackend(bucketName, blobId)
+                        .map(this::toPushbackStream)
+                        .flatMap(pushbackInputStream -> saveInCache(pushbackInputStream, blobId))))
+            .switchIfEmpty(readFromBackend(bucketName, blobId))
```

Comment:
I don't get why we have `readFromBackend` twice

Reply:
It different cases

Reply:
could you please try harder to not have this call twice?

Reply:
Well, i will give it a try